internal/git/gitignore.go

```diff
@@ -0,0 +1,134 @@
+package git
+
+import (
+	"context"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/monochromegane/go-gitignore"
+	"github.com/windmilleng/tilt/internal/logger"
+	"github.com/windmilleng/tilt/internal/ospath"
+)
+
+// Known feature differences from git:
+// 1. does not use git config core.excludesfile
+// 2. only looks for .gitignore in repo root, instead of all directories between dirname(file) and repo root
+// 3. does not use .git/info/exclude
+// 4. does not take index into account
+
+type IgnoreTester interface {
+	IsIgnored(f string, isDir bool) (bool, error)
+}
+
+// an IgnoreTester that ignores nothing
+type falseIgnoreTester struct{}
+
+func (falseIgnoreTester) IsIgnored(f string, isDir bool) (bool, error) {
+	return false, nil
+}
+
+var _ IgnoreTester = falseIgnoreTester{}
+
+// ignores files specified in .gitignore
+type gitIgnoreTester struct {
+	repoRoot      string
+	ignoreMatcher gitignore.IgnoreMatcher
+}
+
+var _ IgnoreTester = gitIgnoreTester{}
+
+func (i gitIgnoreTester) IsIgnored(f string, isDir bool) (bool, error) {
+	_, isChild := ospath.Child(i.repoRoot, f)
```

Comment:
do we care about resolving symlinks here? o_0

Reply:
Good question. I'm gonna TODO that for now.